File: backend/Services/Parsers/ThriceParser.cs
using System.Text.RegularExpressions;
using backend.Models;

namespace backend.Services.Parsers
{
    public class ThriceParser : BasicScoreResultParser
    {
        private ILogger<ThriceParser> _logger;

        public ThriceParser(ILogger<ThriceParser> logger) : base(logger)
        {
            _logger = logger;
        }

        public override bool CountWinner => true;
        public override string GameName => "Thrice";
        public override bool GolfScoring => false;
        public override string? HelpText => null;
        protected override Regex Parser => new Regex($@"{GameName} Game #\d+ → ((?<{ScoreGroup}>[\d]+) points|(?<{ScoreGroup}>I got a perfect score on today's Thrice))");
        protected override string? ExtraContent => Url;
        public override string Url => "https://thricegame.com";

        protected override DailyResult SetScore(DailyResult dailyResult, Match parserResults)
        {
            if (parserResults.Groups.ContainsKey(ScoreGroup) && parserResults.Groups[ScoreGroup].Value == "I got a perfect score on today's Thrice")
            {
                dailyResult.Score = 15;
            }

            return base.SetScore(dailyResult, parserResults);
        }
    }
}

File: lambdas/src/parsers/thrice-parser.ts
import { DailyResult } from '../models/daily-result';
import { BasicScoreResultParser } from './base/basic-score-parser';
import { getGroup, hasGroup } from './base/result-parser';

export class ThriceParser extends BasicScoreResultParser {
  readonly countWinner = true;
  readonly gameName = 'Thrice';
  readonly golfScoring = false;
  readonly helpText = null;
  protected readonly extraContent = this.url;
  readonly url = 'https://thricegame.com';
  
  // Matches "Thrice Game #365 → 3 points" or "Thrice Game #507 → I got a perfect score"
  protected readonly parser = new RegExp(
    `${this.gameName} Game #\\d+ → ((?<${BasicScoreResultParser.SCORE_GROUP}>[\\d]+) points|(?<perfect>I got a perfect score on today's Thrice))`
  );
  
  protected setScore(dailyResult: DailyResult, match: RegExpMatchArray): DailyResult {
    // Check for perfect score first
    if (hasGroup(match, 'perfect') && getGroup(match, 'perfect')) {
      dailyResult.score = 15;
      return dailyResult;
    }
    
    return super.setScore(dailyResult, match);
  }
}
